app/src/main/java/tech/jpco/qen/viewModel/VMDataClasses.kt

```diff
@@ -0,0 +1,44 @@
+package tech.jpco.qen.viewModel
+
+sealed class MetaEvent {
+    object CyclePage : MetaEvent()
+    object ClearPage : MetaEvent()
+    data class NewPage(val ar: Float) : MetaEvent()
+    data class SelectPage(val page: Int) : MetaEvent()
+    data class CurrentPage(val ar: Float) : MetaEvent()
+}
+
+//TODO refactor content and ratio into normal data class implementation?
+data class SelectedPage(val current: Int, val total: Int) {
+    var content: List<DrawPoint> = listOf()
+        private set
+
+    var ratio: Float = 0f
+        private set
+
+    constructor(current: Int, total: Int, content: List<DrawPoint>, ratio: Float) : this(current, total) {
```

Comment:
Maybe a silly design choice, but I wanted to make it so that current and total page numbers were the only criteria for the data-class's autogenerated `equals` and hash methods, and I'm pretty sure excluding them from the default constructor does that. Needless and needlessly convoluted?

Reply:
Ah I see. If that's the requirement, it does work, although it's a bit hacky. I couldn't find a place in the code where the equals or hashcodes are needed, although I probably missed something, so if there's an important reason to leave them out then it makes sense. 

Perhaps if the `content` of the selected page is not important to its identity, then maybe it belongs in a different data model.

Reply:
Can you clarify/give an example of what you mean in the last sentence? One can't inherit _from_ data classes, and a "host" data class having a member variable that was in turn another data class would make the host class's identity dependent on the hosted's, no?